### PR TITLE
#429 - JUL to SLF4J logging

### PR DESCRIPTION
**Pull request by marceloamadeu**
For #429

- [Extension Model](https://junit.org/junit5/docs/current/user-guide/#extensions)
- [Configuration Parameters](https://junit.org/junit5/docs/snapshot/user-guide/#running-tests-build-maven-config-params)
- [Automatic Extension Registration](https://junit.org/junit5/docs/current/user-guide/#extensions-registration-automatic)

### CODE REVIEW

File: eo-runtime/src/main/java/org/eolang/JulToSlf4jExtension.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
+package org.eolang;
```
Comment (yegor256) [inline eo-runtime/src/main/java/org/eolang/JulToSlf4jExtension.java:1]: @marceloamadeu I suggest moving this class to `src/test/java`

File: eo-runtime/pom.xml [hunk 2]
```diff
@@ -34,11 +34,17 @@ SOFTWARE.
   <name>eo-runtime</name>
   <description>EO runtime library</description>
   <dependencies>
+    <dependency>
+      <groupId>org.junit.jupiter</groupId>
+      <artifactId>junit-jupiter-api</artifactId>
+      <version>5.8.1</version>
+      <scope>compile</scope>
```
Comment (yegor256) [inline eo-runtime/pom.xml:41]: @marceloamadeu not a good idea :) Just move the class to `src/test` and you won't need `compile` scope here

**Comment by yegor256** [comment]
@marceloamadeu see my comments

File: eo-runtime/src/main/java/org/eolang/JulToSlf4jExtension.java (hunk 1 above)
Comment (marceloamadeu) [reply eo-runtime/src/main/java/org/eolang/JulToSlf4jExtension.java]: @yegor256 done...

File: eo-runtime/pom.xml (hunk 2 above)
Comment (marceloamadeu) [reply eo-runtime/pom.xml]: Done...

**Comment by yegor256** [comment]
@marceloamadeu ready for review or you are still working on it?

**Comment by marceloamadeu** [comment]
@yegor256 ready... :+1:

File: eo-runtime/pom.xml [hunk 3]
```diff
@@ -34,11 +34,17 @@ SOFTWARE.
...
+      <artifactId>junit-jupiter-api</artifactId>
+      <version>5.8.1</version>
+      <scope>test</scope>
+    </dependency>
     <dependency>
       <groupId>org.slf4j</groupId>
       <artifactId>jul-to-slf4j</artifactId>
       <version>2.0.0-alpha5</version>
-      <scope>test</scope>
+      <scope>compile</scope>
```
Comment (yegor256) [inline eo-runtime/pom.xml:47]: @marceloamadeu still wrong. We shouldn't force our users to use SLF4J. We just use JUL and _they_ decide later what bridge to use.

File: eo-runtime/src/test/java/JulToSlf4jExtension.java [hunk 4]
```diff
@@ -0,0 +1,24 @@
+import org.junit.jupiter.api.extension.BeforeAllCallback;
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.slf4j.bridge.SLF4JBridgeHandler;
+
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import static org.junit.jupiter.api.extension.ExtensionContext.Namespace.GLOBAL;
```
Comment (yegor256) [inline eo-runtime/src/test/java/JulToSlf4jExtension.java:7]: @marceloamadeu `import static` is a bad practice, don't use it

File: eo-runtime/src/test/java/JulToSlf4jExtension.java [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+
+public class JulToSlf4jExtension implements BeforeAllCallback {
+    private static boolean started = false;
+
+    /**
```
Comment (yegor256) [inline eo-runtime/src/test/java/JulToSlf4jExtension.java:12]: @marceloamadeu no need to use Javadoc with methods annotated with `@Override`

File: eo-runtime/src/test/java/JulToSlf4jExtension.java [hunk 6]
```diff
@@ -0,0 +1,24 @@
+import org.junit.jupiter.api.extension.BeforeAllCallback;
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.slf4j.bridge.SLF4JBridgeHandler;
+
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import static org.junit.jupiter.api.extension.ExtensionContext.Namespace.GLOBAL;
+
+public class JulToSlf4jExtension implements BeforeAllCallback {
```
Comment (yegor256) [inline eo-runtime/src/test/java/JulToSlf4jExtension.java:9]: @marceloamadeu here Javadoc is required (see how it's done for other classes)

File: eo-runtime/src/test/java/JulToSlf4jExtension.java [hunk 7]
```diff
@@ -0,0 +1,24 @@
+import org.junit.jupiter.api.extension.BeforeAllCallback;
```
Comment (yegor256) [inline eo-runtime/src/test/java/JulToSlf4jExtension.java:1]: @marceloamadeu see other classes, they all have licenses at the top of the file

File: eo-runtime/src/test/java/JulToSlf4jExtension.java [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+    private static boolean started = false;
+
+    /**
+     * Override the beforeAll() method
+     */
+    @Override
+    public void beforeAll(ExtensionContext context) {
+        if (!started) {
+            started = true;
+            context.getRoot().getStore(GLOBAL).put("JulToSlf4jExtension",this);
```
Comment (yegor256) [inline eo-runtime/src/test/java/JulToSlf4jExtension.java:19]: @marceloamadeu you missed a space char after the comma

**Comment by yegor256** [comment]
@marceloamadeu check again my comments please

File: eo-runtime/src/test/java/JulToSlf4jExtension.java (hunk 4 above)
Comment (marceloamadeu) [reply eo-runtime/src/test/java/JulToSlf4jExtension.java]: @yegor256 removed `import static`...

File: eo-runtime/src/test/java/JulToSlf4jExtension.java (hunk 8 above)
Comment (marceloamadeu) [reply eo-runtime/src/test/java/JulToSlf4jExtension.java]: @yegor256 has been added a space character after the comma.

File: eo-runtime/src/test/java/JulToSlf4jExtension.java (hunk 7 above)
Comment (marceloamadeu) [reply eo-runtime/src/test/java/JulToSlf4jExtension.java:24]: @yegor256 licenses were added.

File: eo-runtime/src/test/java/JulToSlf4jExtension.java (hunk 5 above)
Comment (marceloamadeu) [reply eo-runtime/src/test/java/JulToSlf4jExtension.java:42]: @yegor256 was added. Please verify.

File: eo-runtime/src/test/java/JulToSlf4jExtension.java (hunk 6 above)
Comment (marceloamadeu) [reply eo-runtime/src/test/java/JulToSlf4jExtension.java:39]: @yegor256 was added. Please verify.

File: eo-runtime/pom.xml (hunk 3 above)
Comment (marceloamadeu) [reply eo-runtime/pom.xml:47]: @yegor256 I didn't understand how to do it. Please check that it is correct now.

**Comment by marceloamadeu** [comment]
@yegor256 Please check. Thanks

**Comment by yegor256** [comment]
@rultor merge

**Comment by yegor256** [comment]
@marceloamadeu thanks!

**Pull request merged**